### PR TITLE
Add basic examples

### PR DESCRIPTION
**Pull request by KaQuMiQ**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -39,6 +39,7 @@ dev = [
     "pytest-cov~=4.1.0",
     "pytest-asyncio~=0.23.0",
     "python-dotenv~=1.0.0",
+    "ipykernel~=6.29",
```
Comment (piotr-rarus) [inline pyproject.toml:42]: Add whole `jupyter` dependency.

File: examples/BasicConversation.ipynb [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    "    Toolbox,\n",
+    "    conversation_completion,\n",
+    "    ctx,\n",
+    "    load_env,\n",
+    "    openai_lmm_completion,\n",
+    "    tool,\n",
+    ")\n",
+    "from draive.conversation.message import ConversationMessage\n",
+    "\n",
+    "load_env()  # loads OPENAI_API_KEY from .env file\n",
```
Comment (piotr-rarus) [inline examples/BasicConversation.ipynb:46]: Finish 1st cell here. Only imports and env variables.

File: examples/BasicConversation.ipynb [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    "\n",
+    "\n",
+    "# prepare a basic tool for getting current date and time\n",
+    "@tool(description=\"UTC time and date now\")\n",
```
Comment (piotr-rarus) [inline examples/BasicConversation.ipynb:50]: Tool definition in separate 2nd cell.

File: examples/BasicConversation.ipynb [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    "async def utc_datetime() -> str:\n",
+    "    return datetime.now(UTC).strftime(\"%A %d %B, %Y, %H:%M:%S\")\n",
+    "\n",
+    "\n",
+    "# initialize dependencies and configuration\n",
```
Comment (piotr-rarus) [inline examples/BasicConversation.ipynb:55]: Execution below in final 3rd cell.

File: examples/BasicModelGeneration.ipynb [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    "    Model,\n",
+    "    OpenAIChatConfig,\n",
+    "    OpenAIClient,\n",
+    "    ctx,\n",
+    "    generate_model,\n",
+    "    load_env,\n",
+    "    openai_lmm_completion,\n",
+    ")\n",
+    "\n",
+    "load_env()  # loads OPENAI_API_KEY from .env file\n",
```
[the remark this answers is not in the document]
Comment (piotr-rarus) [reply examples/BasicModelGeneration.ipynb:40]: You can also move narrating bits from comments to markdown cells.

**Review by piotr-rarus — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (KaQuMiQ) [reply pyproject.toml:42]: it seems that it is not required, ipykernel is enough to run in VSCode

Comment (KaQuMiQ) [reply pyproject.toml]: removed the dependency

**Review by piotr-rarus — APPROVED** [review]

**Pull request merged**